doc/rados/operations/crush-map.rst

```diff
@@ -709,6 +709,13 @@ The relevant erasure-code profile properties are as follows:
    [default: ``default``].
  * **crush-failure-domain**: the CRUSH bucket type used in the distribution of
    erasure-coded shards [default: ``host``].
+ * **crush-osds-per-failure-domain**: Maximum number of OSDs to place in each
+   failure domain -- defaults to 1.  Using a value greater than one will
+   cause a CRUSH MSR rule to be created, see below.  Must be specified if
+   crush-num-failure-domains is specified.
+ * **crush-num-failure-domains**: Number of failure domains to map.  Must be
+   specified if crush-osds-per-failure-domain is specified.  Results in
+   a CRUSH MSR rule being created.
```

Comment:
changing back to 1 would lead to creation of a normal CRUSH rule right? Can these values be changed on-the-fly?

Reply:
backticks around crush-num-failure-domains and crush-osds-per-failure-domain

Reply:
"on-the-fly" could mean a few things.  All crush calculations are performed using a specific OSDMap epoch.  Once an OSDMap is published at epoch e, every calculation done at epoch e will use an identical copy of the crush rules and osdmap.

OSDMaps can change between epochs.  AFAIK, there is no user exposed mechanism for changing the erasure-code properties because the implementation cannot handle changing the params related to the number of shards or the algorithm used to encode/decode.  You *can* change the crush rule by editing the crush map and injecting it, which would cause subsequent OSDMap epochs to use your new rule.  However, doing so without extreme care could result in reshuffling all of the data in your cluster which would probably be fatal.

In short, as with every crush rule, you likely do not want to change it once you have created the pool.

Reply:
Added backticks.